src/libServer/WebsocketServer.h

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (C) 2019 Zilliqa
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef ZILLIQA_SRC_LIBSERVER_WEBSOCKETSERVER_H_
+#define ZILLIQA_SRC_LIBSERVER_WEBSOCKETSERVER_H_
+
+#include <json/json.h>
+#include <mutex>
+#include <set>
+#include <unordered_map>
+
+#include "websocketpp/config/asio_no_tls.hpp"
+#include "websocketpp/server.hpp"
+
+#include "common/Constants.h"
+#include "depends/common/FixedHash.h"
+#include "libData/AccountData/Address.h"
+#include "libData/AccountData/TransactionReceipt.h"
+#include "libData/BlockData/Block.h"
+
+typedef websocketpp::server<websocketpp::config::asio> websocketserver;
+
+using IpSocketMap =
+    std::unordered_map<std::string, websocketpp::connection_hdl>;
+
+enum WEBSOCKETQUERY : unsigned int { NEWBLOCK, EVENTLOG };
+
+using EndpointQueryIndex = std::unordered_map<std::string, WEBSOCKETQUERY>;
+
+struct EventLogSocketTracker {
+  // for updating event log for client subscribed
+  std::unordered_map<Address, std::set<std::string>> m_addr_ip_map;
+  // for removing socket from m_eventlog_hdl_tracker
+  std::unordered_map<std::string, std::set<Address>> m_ip_addr_map;
+
+  void remove(const std::string& ip) {
+    auto iter_ha = m_ip_addr_map.find(ip);
+    if (iter_ha == m_ip_addr_map.end()) {
+      return;
+    }
+    for (const auto& addr : iter_ha->second) {
+      auto iter_ah = m_addr_ip_map.find(addr);
+      if (iter_ah != m_addr_ip_map.end()) {
+        iter_ah->second.erase(ip);
+      }
+      if (iter_ah->second.empty()) {
+        m_addr_ip_map.erase(iter_ah);
+      }
+    }
+    m_ip_addr_map.erase(iter_ha);
+  }
+
+  void update(const std::string& ip, const std::set<Address>& addresses) {
+    for (const auto& addr : addresses) {
+      m_addr_ip_map[addr].emplace(ip);
+    }
+    m_ip_addr_map[ip] = addresses;
+  }
+
+  void clean() {
+    m_addr_ip_map.clear();
+    m_ip_addr_map.clear();
+  }
+};
+
+class WebsocketServer : public Singleton<WebsocketServer> {
+ public:
+  /// Returns the singleton AccountStore instance.
+  static WebsocketServer& GetInstance() {
+    static WebsocketServer ws;
+    return ws;
+  }
+
+  /// clean in-memory data structures
+  void clean();
+
+  /// Send string data to hdl connection
+  bool sendData(const websocketpp::connection_hdl& hdl,
```

Comment:
I guess your naming convention is to use lower case for private functions?  In that case, we can move this function to the private section perhaps.